bitsofgood/pencils-for-success
# Endpoint to delete chapter

### Description
Added the endpoint `DELETE /api/chapters/:id` to delete chapter specified by `:id`.
If a chapter has an assigned chapter user, the user is also deleted since a chapter user can only exist for a given chapter.

**Note:** Prisma currently does not support cascading deletes ([source](https://www.prisma.io/docs/guides/database/advanced-database-tasks/cascading-deletes/postgresql#7-introspect-your-database-with-prisma)).
Hence, all the required deletions are done in a single transaction to ensure if an error occurs in one of the deletions, all deletions made earlier are reverted.


**Related Issues/PRs:** 

- Closes #41

### Test Plan
- Delete a chapter without authenticating. It should return a 401 error.
- Delete a chapter without authenticating as an admin. It should return a 401 error.
- Delete a chapter with an invalid chapter id. It should return a 400 error.
- Delete a chapter with a valid chapter id. It should delete the corresponding chapter and its associated chapter user.
